File: cpp/arguments.cpp
#ifndef SRlib_arguments
#define SRlib_arguments

vector <int> int2ints(int idx, vector <int> lengths){
    vector <int> indices; //Indices to return
    int len = 1; //current length to divide by
    for(int i=0;i<lengths.size();i++){
        indices.push_back(idx/len % lengths[i]);
        len *= lengths[i];
    }
    return(indices);
}

#endif

File: R/arguments.R


getArgs <- function(){
    args = commandArgs(trailingOnly=TRUE)
    if(length(args) > 0)
        arg <- as.integer(args[1])
    return(arg)
}

#Convert integer index to many indices (assume from 0)
#idx = single integer index
#lengths = the length (modulus) of the different indices
int2ints <- function(idx,lengths){
    indices <- rep(NA,length(lengths))  #Indices to return
    len <- 1              #Current length to divide by
    for(i in 1:length(lengths)){
        indices[i] <- floor(idx/len) %% lengths[i]
        len <- len * lengths[i]
    }
    return(indices)
}